### ISSUE TITLE
Add the `cache` property when handling CSV-from-URL

### ISSUE THREAD

**Issue by pwalsh** [openspending/openspending#870]
Always set a path for a resource that is a URL, and always copy it into the datastore so it is co-located with the datapackage.

https://github.com/dataprotocols/dataprotocols/issues/223#issuecomment-177813753

**Comment by pwalsh** [comment]
@akariv 

I'm assigning this for Levko, to pick up when / after #873. This is quite important as a pattern for the datastore, before we get too much data.

**Comment by akariv** [comment]
Do we want to depend on this?
Officially we want to be able to support FDPs coming from an external
source (not the Packager), and these might have a URL and not a local file.
Furthermore, there might be a good reason to have a URL and not a local
file (e.g. if we have a constantly updating source of data which we want to
load periodically).

**Comment by pwalsh** [comment]
Hey.

I should have given more context. The original problem I highlight in the comment linked to in the first issue led to the discussion of introducing a `cache` property for data package resources, see https://github.com/dataprotocols/dataprotocols/issues/243 

We don't want to depend on it, no, we want to use it as a fallback IF remote resources disappear (which they will).

As for periodic updating, I agree 100% that it is an awesome possibility (and can also eventually relate to the concept of automated data processing pipelines into OS), but not sure we need to entertain that use case right now.

WDYT?

**Comment by akariv** [comment]
So in essence what you're saying is that we add the `cache` property for
these kind of packages, while keeping the original URL as reference. I'm
okay with that.
BTW, this could be a neat solution for #897 as well.

So - #897 + 3rd party FDPs leads me to think that we might want to do this
on the server (i.e. create a copy of the loaded file, after validation and
transcoding to UTF-8), and then save the amended datapackage.json to the
same place / new place in the datastore.

**Comment by akariv** [comment]
Should be handled along with #923

**Comment by pwalsh** [comment]
@akariv WDYT now? Have with any resolution you suggest.

**Comment by pwalsh** [comment]
@akariv how about now? I assume this is now backed in to DPP

**Comment by sglavoie** [comment]
**WONTFIX/CLEANUP**: Closing old issue without recent activity (>2 years) as part of #1478. Please feel free to re-open if this is still relevant.
